aiohttp/client_reqrep.py

```diff
@@ -290,7 +291,8 @@ def update_body_from_data(self, data):
             if not self.chunked and isinstance(data, io.BytesIO):
                 # Not chunking if content-length can be determined
                 size = len(data.getbuffer())
-                self.headers[hdrs.CONTENT_LENGTH] = str(size)
+                if hdrs.CONTENT_LENGTH not in skip_auto_headers:
+                    self.headers[hdrs.CONTENT_LENGTH] = str(size)
```

Comment:
This one looks wrong. If we don't set content length here, we should fallback to chunked = True mode. Otherwise we'll violate the spec.